service/raft/raft_group_registry.cc

```diff
@@ -377,14 +377,14 @@ future<> raft_group_registry::stop_server(raft::group_id gid, sstring reason) {
 }
 
 future<> raft_group_registry::stop_servers() noexcept {
-    gate g;
+    auto& g = async_gate();
     for (auto it = _servers.begin(); it != _servers.end(); it = _servers.erase(it)) {
         ensure_aborted(it->second, "raft group registry is stopped");
         auto aborted = std::move(it->second.aborted);
-        // discarded future is waited via g.close()
+        // discarded future is waited via g.close() in stop()
         (void)std::move(aborted)->finally([rsfg = std::move(it->second), gh = g.hold()]{});
     }
-    co_await g.close();
+    return make_ready_future();
```

Comment:
Returning a ready future changes semantics of drain_on_shutdown() - it doesn't wait for servers to be stopped. drain_on_shutdown() is not only called from raft_group_registry::stop(), but also directly in main, which is problematic.

Reply:
Not in main - in cql_test_env:

```
            // XXX: drain_on_shutdown raft before stopping the database and
            // query processor. Group registry stop raft groups
            // when stopped, and until then the groups may use
            // the database and the query processor.
            auto drain_raft = defer([this] {
                _group0_registry.invoke_on_all(&service::raft_group_registry::drain_on_shutdown).get();
            });

```

Reply:
I moved the `async_gate().stop()` call to `drain_on_shutdown()` in v2